packages/amplify_analytics_pinpoint/android/src/main/kotlin/com/amazonaws/amplify/amplify_analytics_pinpoint/AmplifyAnalyticsBuilder.kt

```diff
@@ -98,10 +98,13 @@ class AmplifyAnalyticsBuilder {
                         val locationMap = item.value as HashMap<String, String>
                         userProfileBuilder.location(createUserLocation(locationMap))
                     }
-                    "properties" -> {
+                    "propertiesMap" -> {
                         val propertiesMap = item.value as HashMap<String, Any>
                         userProfileBuilder.customProperties(createAnalyticsProperties(propertiesMap))
                     }
+                    "propertiesTypesMap" -> {
+                        // Ignore, we can infer types just using propertiesMap
+                    }
```

Comment:
probably a good idea to do it anyways? I've seen in DataStore `1` being translated to `1.0` after traveling through method channel serialization. If that didn't happen in your testing, then we can keep it as is.

Reply:
hmm what are you referring to?  We ignore this case as it's already handled for the propertiesMap case.  

I've verified that for Android a double property sent as 10.0 and an int property sent as 10 are saved as their proper types.

Reply:
iOS was a bit trickier hence the extra data structure propertiesTypesMap being sent from Dart

Reply:
I was referring to inferring the types in android. There were some issues in DataStore when `int` were being converted to `double` by GSON library. If that's not the case here, then we can ignore it.